### ISSUE TITLE
Input 0 is incompatible with layer lstm_LSTM1: expected ndim=3, found ndim=4

### ISSUE THREAD

**Issue by hugesswnn** [tensorflow/tfjs#482]
does not support different dims?  
xs shape [ 1, 2, 1 ]
ys shape [ 1, 3, 1 ]

```javascript
(async () => {
    var xs = tf.tensor3d([
        [[8],[3]], //
    ]);

    var ys = tf.tensor3d([
        [[0],[1],[0]],
    ]);

    const model = tf.sequential();

    const hidden = tf.layers.lstm({
        units: 3,
        activation: 'sigmoid',
        inputShape: [3,1],
        returnSequences: true
    });
    model.add(hidden);

    //output layer
    const output = tf.layers.lstm({
        units: 1,
        activation: 'sigmoid',
        returnSequences: true
    })
    model.add(output);

    //compile
    const sgdoptimizer = tf.train.sgd(0.0001);
    model.compile({
        optimizer: sgdoptimizer,
        loss: tf.losses.meanSquaredError
    });
    try{
        await model.fit(xs, ys, { epochs: 30 });
        const inputs = tf.tensor3d([[[1],[2]]]);
        let outputs = model.predict(inputs);
        console.log(outputs.argMax().dataSync());  
    }catch(err){
        console.log(err);
    }
})()

**Comment by xcapt** [comment]
what was the solution to close the issue in the code above ?
